psl-core/src/main/java/org/linqs/psl/util/ArrayUtils.java

```diff
@@ -26,6 +27,27 @@
     // Static only.
     private ArrayUtils() {}
 
+    public static double[] ensureCapacity(double[] array, int capacity) {
+        if (array.length  <= capacity) {
+            array = Arrays.copyOf(array, (capacity + 1) * 2);
+        }
+        return array;
+    }
+
+    public static float[] ensureCapacity(float[] array, int capacity) {
+        if (array.length  <= capacity) {
+            array = Arrays.copyOf(array, (capacity + 1) * 2);
+        }
+        return array;
+    }
+
+    public static int[] ensureCapacity(int[] array, int capacity) {
+        if (array.length  <= capacity) {
+            array = Arrays.copyOf(array, (capacity + 1) * 2);
+        }
+        return array;
+    }
+
```

Comment:
Add one more for general classes:
`public static <T> T[] ensureCapacity(T[] array, int capacity) {`